Test-EwsEndpoint.ps1

```diff
@@ -0,0 +1,166 @@
+    )
+
+    # Remove variable $BTUserCredential if it exists.
+    if($BTUserCredential)
+    {
+        Remove-Variable BTUserCredential
+    }
+
+    # Ask for credentials.
+    try
+    {
+        $BTUserCredential = Get-Credential -Message "Enter you BitTitan credentials"
+    }
+    catch
+    {
+        Write-Host -ForegroundColor Red  "Error. Get credentials failed with $_"
+        Return;
+    }
+
+    # Request ticket with elevated privilage.
+    try
+    {
+        $ticket = Get-MW_Ticket -Credentials $BTUserCredential -ElevatePrivilege:$true
+    }
+    catch
+    {
+        Write-Host -ForegroundColor Red  "Error. Getting MigationWiz ticket failed with $_"
+        Return;
+    }
+
+    # Get mailbox connector based on $projectID provided.
+    try
+    {
+        $mailboxConnector = Get-MW_MailboxConnector -Ticket $ticket -Id $projectID -ShouldUnmaskProperties:$true
+    }
+    catch
+    {
+        Write-Host -ForegroundColor Red  "Error. Getting mailbox connector failed with $_"
+        Return;
+    }
+
+    # Check if ProjectType is Mailbox or Archive
+    if($($mailboxconnector.ProjectType) -ne "Mailbox" -and $($mailboxconnector.ProjectType) -ne "Archive ")
+    {
+        Write-Host "$($mailboxConnector.Name) is a $($mailboxConnector.ProjectType) project, not a Mailbox project."
+        Return;
+    }
+
+    # Gather Export credentials
+    $secureExportPassword = $($mailboxConnector.ExportAdministrativePassword) | ConvertTo-SecureString -AsPlainText -Force
+    $secureExportCreds = New-Object System.Management.Automation.PSCredential -ArgumentList $($mailboxConnector.ExportAdministrativeUserName), $secureExportPassword
+
+    # Determine whether Office 365 or customer EWS URL will be used to test connectivity against
+    if($mailboxConnector.ExportType -eq "ExchangeOnline2")
+    {
+        $exportUrl = 'https://outlook.office365.com/EWS/Exchange.asmx'
+    }
+    else
+    {
+        $exportUrl = $mailboxConnector.ExportConfiguration.Url
+    }
+
+    # Gather Import credentials
+    $secureImportPassword = $($mailboxConnector.ImportAdministrativePassword) | ConvertTo-SecureString -AsPlainText -Force
+    $secureImportCreds = New-Object System.Management.Automation.PSCredential -ArgumentList $($mailboxConnector.ImportAdministrativeUserName), $secureImportPassword
+
+    # Determine whether Office 365 or custome EWS URL will be used to test connectivity against
+    if($mailboxConnector.ImportType -eq "ExchangeOnline2")
+    {
+        $importUrl = 'https://outlook.office365.com/EWS/Exchange.asmx'
+    }
+    else
+    {
+        $importUrl = $mailboxConnector.ImportConfiguration.Url
+    }
+
+    # Run EWS URL connectivity check
+    Write-Host "Export EWS URL Test against $exportUrl"
+    EWSUrlTest -EWS_URL $exportUrl -Credentials $secureExportCreds
+    Write-Host "Import EWS URL Test against $importUrl`n`n"
+    EWSUrlTest -EWS_Url $importUrl -Credentials $secureImportCreds
+
+}
+
+Test-EwsEndpoints

```

Comment:
You set this function to use a mandatory parameter, "projectID", but you do not pass in a projectID when you call the function here.